src/__tests__/recursive-input.test.ts

```diff
@@ -0,0 +1,588 @@
+import { DocumentNode, GraphQLSchema, parse, validate } from "graphql";
+import { makeExecutableSchema } from "graphql-tools";
```

Comment:
You can use `@graphql-tools/schema` instead because npm package `graphql-tools` has been deprecated :)